main.go

```diff
@@ -14,15 +14,47 @@ import (
 )
 
 const (
+	// Go glog logs
 	glog = `^\S+ \S+ \S+\s+\d+ (?P<file>.+):\d+\] (?P<msg>(?P<msgcore>.*?:)?.*)`
+
+	// Nginx access logs
+	// 1.2.3.4 - - [18/Feb/2016:10:05:22 +0000] "GET /client/assets/some.css HTTP/1.1" 200 1090 "https://somereferer" "Mozilla/5.0 (Linux; Android 5.1.1; D6603 Build/23.4.A.1.264; wv) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/47.0.2526.100 Mobile Safari/537.36 (Mobile; afma-sdk-a-v8489000.7327000.1)"
+	nginx = `^(?P<remote>[^ ]*) (?P<host>[^ ]*) (?P<user>[^ ]*) \[[^\]]*\] "(?P<method>\S+)(?: +(?P<path>[^\"]*) +\S*)?" (?P<code>[^ ]*) (?P<size>[^ ]*)(?: "(?P<referer>[^\"]*)" "(?P<agent>[^\"]*)")?$`
+
+	// Nginx error logs
+	// 2016/02/18 10:06:44 [crit] 13679#0: *3683 SSL_do_handshake() failed (SSL: error:140A1175:SSL routines:SSL_BYTES_TO_CIPHER_LIST:inappropriate fallback) while SSL handshaking, client: 1.2.3.4, server: 0.0.0.0:443
+	nginxerror = `^.* \[(?P<level>\w+)\] (?P<msg>.*)$`
+
+	// Rails multiline. We don't deal with these nicely.
+	// I, [2016-02-18T11:27:36.879342 #14224]  INFO -- : Completed 200 OK in 141ms (Views: 95.1ms | ActiveRecord: 9.7ms)
+	// I, [2016-02-18T11:27:36.882539 #14200]  INFO -- : Started PATCH "/some/url" for 1.2.3.4 at 2016-02-18 11:27:36 +0000
+	// F, [2016-02-18T06:52:19.155825 #30782] FATAL -- :
+	// PG::ConnectionBad (could not connect to server: Connection refused
+	//     Is the server running on host "localhost" (127.0.0.1) and accepting
+	//     TCP/IP connections on port 5432?
+	// ):
+	//   vendor/bundle/ruby/2.1.0/gems/activerecord-4.0.2/lib/active_record/connection_adapters/postgresql_adapter.rb:831:in `initialize'
+	rails = `^[DIEF], \[.*\]\s+(?P<level>\w+)\s+-- : (?P<msg>.*)$`
```

Comment:
The part [...] is actually [${timestamp_ISO8601} #${pid}] in case you want to improve that.